client/src/components/ConfigTemplates/SourceOptionCard.vue

```diff
@@ -17,40 +23,84 @@ type OptionType = {
 
 interface Props {
     /** The source option to display */
-    sourceOption: FileSourceTemplateSummary | ObjectStoreTemplateSummary;
+    sourceOption: SourceOption;
     /** The route to navigate to when the user selects this option */
-    selectRoute: string;
+    selectRoute?: string;
     /** The type of the source option */
-    optionType: OptionType;
+    optionType?: OptionType;
     /** Whether to display the card in a grid view */
     gridView?: boolean;
+    /** Whether the card is selected */
+    selected?: boolean;
+    /** The title of the submit button */
+    submitButtonTooltip?: string;
+    /** Whether to show the selection mode */
+    selectionMode?: boolean;
 }
 
 const props = defineProps<Props>();
 
-const description = markup(props.sourceOption.description ?? "", true) ?? undefined;
+const emit = defineEmits<{
+    (e: "select", sourceOption: SourceOption): void;
+}>();
+
+const uniqueId = computed(() => {
+    if ("id" in props.sourceOption) {
+        return props.sourceOption.id;
+    } else if ("object_store_id" in props.sourceOption && props.sourceOption.object_store_id) {
+        return props.sourceOption.object_store_id;
+    } else {
+        return Math.random().toString(36).substring(7);
```

Comment:
Can you add a comment here to explain what you're doing here ?

Reply:
@mvdbeek The `sourceOption` has an id if it's an instance of `FileSourceTemplateSummary` or `ObjectStoreTemplateSummary`, but `object_store_id` can be `undefined` in `UserConcreteObjectStoreModel`.
This `id` used to be required before [this change](https://github.com/galaxyproject/galaxy/pull/20064/files#diff-5e6ecdc70b61f43fa634318c28f367814cdb95867d48602f572bd8d67612d43bL19-R21), but now `GCard` auto-generates it if it’s missing, so this line can be removed. Thanks for pointing this out!